### DISCUSSION TITLE
Feature request: Deselect all

### DISCUSSION THREAD

**Discussion by quelsemme** [AlexW00/obsidian-note-linker#27] [Ideas]
As someone who installed the plugin with 700 notes, the plugin actually crashes while I'm trying to go through and select all of the options (and still, I think I'd need a few days to do it all). That makes the 'Process Vault' function useless to me. Unless I've gone through the whole list, I'm not going to risk hitting submit. 

One solution would be that nothing is selected automatically when you Process the vault, that way I could work through for however long I had, hit submit and then know it was only going to change things I decided to link. It's just too risky to assume it's going to do the right thing for everything (particularly because it tends to scan something like 'Walter Benjamin' and suggest as options Walter Benjamin, Benjamin, Benjaminian. If I'm not precise and measured with my timing, I end up with a link that looks like [[Walter [[Walter Benjamin|Benjamin]]|Walter Benjamin]] from clashing single and multiple part links.

As it is, I love the plugin, but I am only using it to a limited capacity, on single notes that I open. I wish I could link my entire existing vault.

**Comment by annekaelber** [comment]
I +1 this idea, for the same reason. I've also suggested a "link first occurrence only" option and a per note approval button. Any one of these ideas would make this process less frustrating, And my vault is just under 800 notes.

**Reply by quelsemme** [reply]
Wonder if @AlexW00 has seen this.

My issue is, 'Scan Vault' scans 1300 notes at the moment, and this number will only increase. You can imagine therefore how long the list of unlinked mentions is across that vault. It's impossible for me to go through the entire list and check it, and too risky to just 'accept all' (also because anytime a word appears in a hashtag, it also gets swept up into Note Linker which breaks all of my tags by throwing a link in the middle of a hashtag)

An alternative fix would be a 'Scan Folder' option as a middle ground between 'Scan Note' and 'Scan Vault'
